packages/qwik/src/core/render/dom/render.public.ts

```diff
@@ -23,23 +23,32 @@ export interface RenderOptions {
   serverData?: Record<string, any>;
 }
 
+/**
+ * @alpha
+ */
+export interface RenderCleanupFn {
+  (): void;
+}
+
 /**
  * Render JSX.
  *
  * Use this method to render JSX. This function does reconciling which means
  * it always tries to reuse what is already in the DOM (rather then destroy and
  * recreate content.)
+ * It returns a cleanup function you could use for cleaning up subscriptions.
  *
  * @param parent - Element which will act as a parent to `jsxNode`. When
  *     possible the rendering will try to reuse existing nodes.
  * @param jsxNode - JSX to render
+ * @returns An async cleanup function
  * @alpha
  */
 export const render = async (
   parent: Element | Document,
   jsxNode: JSXNode | FunctionComponent<any>,
   opts?: RenderOptions
-): Promise<void> => {
+): Promise<RenderCleanupFn> => {
```

Comment:
dont return a function directly, this will end up in breaking changes whenever we want to add something more, instead return an object, which includes a cleanup() method

Reply:
thanks, done